Userland/Applications/Calendar/AddEventDialog.cpp

```diff
@@ -92,8 +94,17 @@ AddEventDialog::AddEventDialog(Core::DateTime date_time, Window* parent_window)
     button_container.add_spacer().release_value_but_fixme_should_propagate_errors();
     auto& ok_button = button_container.add<GUI::Button>("OK"_short_string);
     ok_button.set_fixed_size(80, 20);
-    ok_button.on_click = [this](auto) {
-        dbgln("TODO: Add event icon on specific tile");
+    ok_button.on_click = [&](auto) {
+        JsonObject event;
+        event.set("dtstart", DeprecatedString::formatted("{}-{:0>2d}-{:0>2d}", m_date_time.year(), m_date_time.month(), m_date_time.day()));
```

Comment:
Let's try not to add any more `DeprecatedString`s if we can avoid it (in this case we can).

Since `JsonValue` can be a `StringView`, we can use `String` too.
https://github.com/SerenityOS/serenity/blob/d8a1591722fc91fae48201cf2e1a8e0040ffc512/AK/JsonValue.h#L59-L62

However, to move to `String`, we'd like to propagate errors during creation using the `TRY` macro, so you should move the `on_click` logic into a private function of `AddEventDialog`.

```cpp
ErrorOr<void> add_event_to_calendar()
{
    JsonObject event;
    auto start_date = TRY(String::formatted("{}-{:0>2d}-{:0>2d}", m_date_time.year(), m_date_time.month(), m_date_time.day()));
    event.set("dtstart", start_date);
}
```

Moving this to an `ErrorOr<T>` function also lets you wrap the `calendar.add_event` call further down in a `TRY()` macro.

```cpp
...
TRY(calendar.add_event(event));
calendar.set_dirty(true);
```